slidez-core/src/main/java/com/binarystudio/academy/slidez/app/user/UserController.java

```diff
@@ -1,16 +1,54 @@
 package com.binarystudio.academy.slidez.app.user;
 
+import com.binarystudio.academy.slidez.domain.user.UserService;
+import com.binarystudio.academy.slidez.domain.user.dto.UserDetailsDto;
+import com.binarystudio.academy.slidez.domain.user.mapper.UserMapper;
+import com.binarystudio.academy.slidez.domain.user.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.util.Optional;
+import java.util.UUID;
+
 @RestController
 @RequestMapping("${v1API}/users")
 public class UserController {
 
-	@GetMapping
-	public String hello() {
-		return "Users hello";
-	}
+    @Autowired
+    private UserService userService;
+
+    @GetMapping("userInfo")
+    public ResponseEntity<Object> getByToken(@RequestParam("token") String token) {
+        if(token == null || token.isEmpty()) {
+            return new ResponseEntity<>("Invalid token", HttpStatus.BAD_REQUEST);
+        }
+
+        Optional<UserDetailsDto> userDetailsDto = userService.findByToken(token);
+        if(userDetailsDto.isEmpty()) {
+            return new ResponseEntity<>("Bad token.", HttpStatus.BAD_REQUEST);
+        }
+
+        return new ResponseEntity<>(userDetailsDto.get(), HttpStatus.OK);
+    }
```

Comment:
Why do you merge all your branches to the same one? Or why you have get by token in getById task?

Reply:
Because I've add some fixes together with this part.